spring-cloud-kubernetes-client-config/src/main/java/org/springframework/cloud/kubernetes/client/config/KubernetesClientConfigMapPropertySource.java

```diff
@@ -58,6 +59,12 @@ public KubernetesClientConfigMapPropertySource(CoreV1Api coreV1Api, String name,
 					.getItems().stream().filter(cm -> names.contains(cm.getMetadata().getName()))
 					.forEach(map -> result.putAll(processAllEntries(map.getData(), environment)));
 
+			if (!"".equals(prefix)) {
```

Comment:
My idea is that a `NormalizedSource` _must_ generate a prefix - be that empty or with an actual value. If it is anything other than an empty String, it means one exists and must be used to generate the new property names.